books/projects/shnf/top.lisp

```diff
@@ -22,15 +22,28 @@
 
 ;; A polynomial term over a list of variables:
 
+(defn distinct-symbols (vars)
+  (if (consp vars)
+      (and (distinct-symbols (cdr vars))
+           (symbolp (car vars))
+           (not (member (car vars) (cdr vars))))
+    (null vars)))
+
 (defun polyp (x vars)
+  (declare (xargs :guard (distinct-symbols vars)))
```

Comment:
Change body of this function so that `len` test ensures that `car` and `cdr` are defined.